irksome/stepper.py

```diff
@@ -56,24 +57,49 @@ def __init__(self, F, butcher_tableau, t, dt, u0, bcs=None,
         self.solver = NLVS(problem, solver_parameters=solver_parameters)
 
         if self.num_stages == 1 and self.num_fields == 1:
-            self.ks = (stages,)
+            self.ws = (stages,)
         else:
-            self.ks = stages.split()
+            self.ws = stages.split()
+
+        A1, A2 = splitting(butcher_tableau.A)
+        try:
+            self.updateb = numpy.linalg.solve(A2.T, butcher_tableau.b)
+        except numpy.linalg.LinAlgError:
+            raise NotImplementedError("A=A1 A2 splitting needs A2 invertible")
+        boo = numpy.zeros(self.updateb.shape, dtype=self.updateb.dtype)
+        boo[-1] = 1
+        if numpy.allclose(self.updateb, boo):
+            self._update = self._update_A2Tmb
+        else:
+            self._update = self._update_general
 
-    def _update(self):
+    def _update_general(self):
         """Assuming the algebraic problem for the RK stages has been
         solved, updates the solution.  This will not typically be
         called by an end user."""
-        b = self.butcher_tableau.b
+        b = self.updateb
         dtc = float(self.dt)
         u0 = self.u0
         ns = self.num_stages
         nf = self.num_fields
 
-        ks = self.ks
+        ws = self.ws
         for s in range(ns):
             for i, u0d in enumerate(u0.dat):
-                u0d.data[:] += dtc * b[s] * ks[nf*s+i].dat.data_ro
+                u0d.data[:] += dtc * b[s] * ws[nf*s+i].dat.data_ro
+
+    def _update_A2Tmb(self):
+        """Assuming the algebraic problem for the RK stages has been
+        solved, updates the solution.  This will not typically be
+        called by an end user."""
```

Comment:
I feel like this needs a slightly more pointed docstring - isn't the point here that this is _just_ the special case for `w = Ak`?

Reply:
Yes, I copied/pasted but didn't update it.